src/common/grpc/src/flight.rs

```diff
@@ -50,8 +50,12 @@ pub struct FlightEncoder {
 
 impl Default for FlightEncoder {
     fn default() -> Self {
+        let write_options = writer::IpcWriteOptions::default()
+            .try_with_compression(Some(arrow::ipc::CompressionType::LZ4_FRAME))
+            .unwrap();
```

Comment:
Always `Ok` because the default option has `metadata_version = MetadataVersion::V5`.

But I'm a bit hesitate if we should hard code the compression type here, while I agree that we don't need to make it configurable. Fortunately, the receiver (reader) side would determinate the compression type and do the necessary decompresson work.